app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -151,40 +158,34 @@ class BluetoothViewModel @Inject internal constructor(
                 viewModelScope.launch {
                     val mappingResult = uiStateMapper.mapMessagesAction(action.message.action)
                     if (mappingResult.isSuccess) {
+                        val messageId = action.message.id
                         when (val mappedAction = mappingResult.getOrNull()!!) {
-                            is MessagesAction.HealthSummaryAction -> { /* TODO */
+                            is MessagesAction.HealthSummaryAction -> {
+                                handleHealthSummaryAction(messageId)
                             }
 
                             is MessagesAction.MeasurementsAction -> {
-                                bottomSheetEvents.emit(BottomSheetEvents.Event.DoNewMeasurement)
+                                appScreenEvents.emit(AppScreenEvents.Event.DoNewMeasurement)
                             }
 
-                            is MessagesAction.MedicationsAction -> { /* TODO */
+                            is MessagesAction.MedicationsAction -> {
+                                appScreenEvents.emit(
+                                    AppScreenEvents.Event.NavigateToTab(
+                                        BottomBarItem.MEDICATION
+                                    )
+                                )
                             }
 
                             is MessagesAction.QuestionnaireAction -> { /* TODO */
                             }
 
                             is MessagesAction.VideoSectionAction -> {
                                 viewModelScope.launch {
-                                    engageEducationRepository.getVideoBySectionAndVideoId(
-                                        mappedAction.videoSectionVideo.videoSectionId,
-                                        mappedAction.videoSectionVideo.videoId
-                                    ).getOrNull()?.let { video ->
-                                        navigator.navigateTo(
-                                            EducationNavigationEvent.VideoSectionClicked(
-                                                video = video
-                                            )
-                                        )
-                                    }
+                                    handleVideoSectionAction(mappedAction)
                                 }
                             }
                         }
-                        val messageId = action.message.id
                         messageRepository.completeMessage(messageId = messageId)
```

Comment:
I have created an issue #90 for this because the actions for Questionnaire are still missing in another unmerged branch. Basically, the complete message has little influence in some cases. The messages are removed on the backend side. For example, the message for the healthsummary is only removed 1 day after the meeting, regardless of whether it was marked as complete. nevertheless, I think that it makes sense to execute the complete only when it has been done without errors.